File: src/app/api/models/Candidature.ts
import { Evaluation } from './evaluation';
import { Recrutement } from './recrutement';

export interface Candidature {
  id?: number;
  nom?: String;
  prenom?: String;
  email?: String;
  telephone?: String;
  adresse?: String;
  prochaineAction?: String;
  dateEntretien1?: String;
  dateEntretien2?: String;
  dateEntretien3?: String;
  stadeDeRecrutement?: String;
  moyenne?: number;
  apreciationGlobale?: string;
  recrutement?: Recrutement;
  evaluation?: Evaluation;
  estRetenu?: boolean;
}

export interface CandidatureCreation {
  id?: number;
  nom: String;
  prenom: String;
  email: String;
  telephone: String;
  adresse: String;
  dateEntretien1: string;
  notePresentation?: number;
  noteExperience?: number;
  noteCompetenceEtAtout?: number;
  noteSavoirEtre?: number;
  noteQualiteEtDefaut?: number;
  prochaineAction: String;
  apreciationGlobale: string;
  moyenne?: number;
  estRetenu?: boolean;
}

File: src/main/java/com/safer/RH/models/Candidature.java
package com.safer.RH.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Data
@Entity
public class Candidature {
    @Id
    @GeneratedValue(strategy = jakarta.persistence.GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private String prenom;
    private String email;
    private String telephone;
    private String adresse;
    private String prochaineAction;
    private LocalDate dateEntretien1;
    private double notePresentation;
    private double noteExperience;
    private double noteCompetenceEtAtout;
    private double noteSavoirEtre;
    private double noteQualiteEtDefaut;
    private double moyenne;
    private String  apreciationGlobale;
    @ManyToOne
    private Recrutement recrutement;
    @OneToOne
    private Evaluation evaluation;
    private boolean estRetenu = false;

}

